src/main/java/de/dennisguse/opentracks/data/ContentProviderUtils.java

```diff
@@ -228,6 +228,49 @@ public List<Track> getTracks(ContentProviderSelectionInterface selection) {
         return tracks;
     }
 
+    public List<Track> searchTracks(String searchQuery) {
```

Comment:
Should be a cursor as otherwise all tracks will be loaded (even the one not visible on the screen), right?

Reply:
We could do it multiple ways. Either we use a `CursorLoader`, `Track.Id`s or the full list of tracks.
I did a quick memory dump and found out that one instance of Track takes up ~320 bytes of memory. So that isn't the biggest concern for me. We mainly use RecyclerView to cut down the amount of Views, which take up more memory.
But I can rewrite it to use less memory, if you want me to.

Reply:
It is not an issue for newer devices, but my good old S4 is slow and the memory is limited. Especially, for people who have quite some amount of tracks (1000+), this will help quite a lot.

Reply:
I switched to `List<Track.Id>`, where each id only takes up 16 bytes. To improve this further, we would need to write quite a lot of memory management code.

Reply:
That's not what I meant ;)
IIRC The old implementation was using the data directly from the database (via the cursor using ResourceCursorAdapter).
So, it only loaded the data from the database that was actually shown.
The change would be that `TrackListAdapter.swapData()` is changed use the cursor created by `ContentProviderUtils.searchTracks()` (instead of using a list with the resolved TrackId/data).

Reply:
Sorry for the delay and also sorry for not researching what a Cursor actually is. Feel free to tell me, if you find other things that can be improved

Reply:
No worries :)